modules/social_features/social_group/modules/social_group_request/src/Form/GroupRequestMembershipRequestForm.php

```diff
@@ -2,90 +2,46 @@
 
 namespace Drupal\social_group_request\Form;
 
-use Drupal\Core\Cache\CacheTagsInvalidatorInterface;
-use Drupal\Core\Entity\EntityTypeManagerInterface;
-use Drupal\Core\Form\FormBase;
+use Drupal\Core\Entity\ContentEntityForm;
 use Drupal\Core\Form\FormStateInterface;
-use Drupal\Core\Session\AccountInterface;
-use Drupal\Core\StringTranslation\TranslationInterface;
 use Drupal\grequest\Plugin\Group\Relation\GroupMembershipRequest;
 use Drupal\group\Entity\GroupInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
  * Provides a form to request group membership.
  */
-class GroupRequestMembershipRequestForm extends FormBase {
+class GroupRequestMembershipRequestForm extends ContentEntityForm {
 
   /**
-   * Group entity.
+   * The module handler.
    *
-   * @var \Drupal\group\Entity\GroupInterface
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
    */
-  protected $group;
+  protected $moduleHandler;
 
   /**
-   * Group membership request.
-   *
-   * @var \Drupal\group\Entity\GroupRelationshipInterface
-   */
-  protected $groupContent;
-
-  /**
-   * The cache tags invalidator.
-   *
-   * @var \Drupal\Core\Cache\CacheTagsInvalidatorInterface
-   */
-  protected $cacheTagsInvalidator;
-
-  /**
-   * The current user.
-   *
-   * @var \Drupal\Core\Session\AccountInterface
-   */
-  protected $currentUser;
-
-  /**
-   * Entity type manger.
+   * The entity type manager.
    *
    * @var \Drupal\Core\Entity\EntityTypeManagerInterface
    */
   protected $entityTypeManager;
 
   /**
-   * GroupRequestMembershipRejectForm constructor.
-   *
-   * @param \Drupal\Core\Cache\CacheTagsInvalidatorInterface $cache_tags_invalidator
-   *   The cache tags invalidator.
-   * @param \Drupal\Core\Session\AccountInterface $current_user
-   *   The current user.
-   * @param \Drupal\Core\StringTranslation\TranslationInterface $string_translation
-   *   The string translation.
-   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
-   *   The entity type manager.
+   * {@inheritdoc}
    */
-  public function __construct(
-    CacheTagsInvalidatorInterface $cache_tags_invalidator,
-    AccountInterface $current_user,
-    TranslationInterface $string_translation,
-    EntityTypeManagerInterface $entity_type_manager
-  ) {
-    $this->cacheTagsInvalidator = $cache_tags_invalidator;
-    $this->currentUser = $current_user;
-    $this->setStringTranslation($string_translation);
-    $this->entityTypeManager = $entity_type_manager;
+  public static function create(ContainerInterface $container) {
+    $form = parent::create($container);
+    $form->moduleHandler = $container->get('module_handler');
+    $form->entityTypeManager = $container->get('entity_type.manager');
+    return $form;
```

Comment:
A static `create` function shouldn't do anything beyond calling `__construct`. Not doing so can lead to services being needed which can not be injected through a constructor which blocks testing with PHPUnit.

Additionally in this case `moduleHandler` and `entityTypeManager` are both already provided by `EntityForm` so we can likely remove them from the class altogether.